chromium_src/third_party/blink/renderer/core/loader/mixed_content_checker.h

```diff
@@ -0,0 +1,31 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_CHROMIUM_SRC_THIRD_PARTY_BLINK_RENDERER_CORE_LOADER_MIXED_CONTENT_CHECKER_H_
+#define BRAVE_CHROMIUM_SRC_THIRD_PARTY_BLINK_RENDERER_CORE_LOADER_MIXED_CONTENT_CHECKER_H_
+
+#include "third_party/abseil-cpp/absl/types/optional.h"
+
+// We are hiding IsMixedContent(const String& origin_protocol, const KURL&)
+// because we want to enforce mixed content checks on .onion origins.
+// Publically available protocol-only overload of this method allows to skip
+// .onion specific checks which we don't want.
+
+#define UpgradeInsecureRequest                                              \
```

Comment:
could you also add a test case that upgradeinsecurerequest is working as expected in tor windows?

1. go to onion site over HTTP which includes `<img src='http://mixed.badssl.com/image.jpg'>`
2. the img should be allowed because it can be auto-upgraded to HTTPS